### ISSUE TITLE
Refactor ent macro attributes

### ISSUE THREAD

**Issue by chipsenkbeil** [chipsenkbeil/entity-rs#18]
Currently, we have attributes like the following:

```rust
#[ent(no_builder, no_query, no_typed_methods)]
```

These seem almost pointless (not quite). It may be okay to keep them, but the main ones we want to support are the following:

```rust
#[ent(builder = "name", query = "name")]
```

Additionally, for enums, we need a way to specify the list of types and that the enum wraps them once #15 is complete. This would be at the struct field level:

```rust
#[ent(edge(enum = "MyEnum", types = ["T1", "T2", "T3"]))]
```

The above would change how the load_<edge> method works as it would load the untyped version and attempt to case to each of the given types, wrapping the type in the enum using a call to Into<MyEnum> for T1, T2, or T3. None would be returned when another, unexpected type is found.

This still depends on MyEnum implementing some query, which may not be desirable. To that end, there are also two options for the queries: `#[ent(edge(no_query))]` to remove the query methods on the edge and `#[ent(edge(untyped_query))]` to change the query method to be a generic `entity::Query` instead of a typed version.

**Comment by chipsenkbeil** [comment]
Handled the wrapper type. Going to make a new task about splitting up derives.
